File: Python/BEGINNER/2709 The Coins of Robbie.py
while True:
    try:
        coinList = []
        sumCoins = count = 0

        coins = int(input())
        for _ in range(coins):
            coinList.append(int(input()))

        jump = int(input())
        for i in range(coins-1, -1, -jump):
            sumCoins += coinList[i]

        for number in range(1, sumCoins + 1):
            if sumCoins % number == 0:
                count += 1

        if count == 2:
            print('You’re a coastal aircraft, Robbie, a large silver aircraft.')
        else:
            print('Bad boy! I’ll hit you.')
    except EOFError:
        break

File: C/BEGINNER/2709 The Coins of Robbie.c
#include <stdio.h>

void checkPrime(int sum)
{
    int count=0;
    for (int i=2; i <= sum; i++)
    {
        if (sum % i == 0)
            count += 1;
    }
    if (count == 1)
        printf("You’re a coastal aircraft, Robbie, a large silver aircraft.\n");
    else
        printf("Bad boy! I’ll hit you.\n");
}

int main()
{
    int numberOfCoins, jump;
    
    while (scanf("%d", &numberOfCoins) != EOF)
    {    
        int coinPot[numberOfCoins];
        for (int i=(numberOfCoins-1); i >= 0; i--)
        {
            scanf("%d", &coinPot[i]);
        }
        scanf("%d", &jump);
        int sum=0;
        for (int i=0; i < numberOfCoins; i+=jump)
        {
            sum += coinPot[i];
        }
        checkPrime(sum);
    }
    return 0;
}